src/parsing_routines_matrixnetworks.jl

```diff
@@ -122,3 +122,103 @@ function loadrxnetwork(ft::MatrixNetwork,
 
     ParsedReactionNetwork(rn, nothing)
 end
+
+
+"""
+    For loading networks with complex stoichiomatix ,incidence matrix.
+
+Notes:
+- Assumed that column of complex stoichiomatrix represents composition of reaction complexes,
+  with positive entries of size num_of_species x num_of_complexes, where
+  the non-zero positive entries in the kth column denote stoichiometric
+  coefficients of the species participating in the kth reaction complex.
+
+- The complex incidence matrix, is number of complexes by number of reactions with
+  Bᵢⱼ = -1, if the i'th complex is the substrate of the j'th reaction,
+         1, if the i'th complex is the product of the j'th reaction,
+         0, otherwise
+"""
+#  for dense matrices
+function LoadReacCompNetwork(rateexprs::AbstractVector,
+                        compstoichmat::AbstractMatrix,
+                        incidencemat::AbstractMatrix;
+                        species::AbstractVector = Any[],
+                        params::AbstractVector = Any[])
+
+    numspecs, numcomp = size(compstoichmat)
+    @assert all(>=(0),compstoichmat)
+    @assert numcomp == size(incidencemat, 1)
+    @assert all(∈([-1,0,1]),incidencemat)
+    numrxs = size(incidencemat, 2)
+
+    ModelingToolkit.@parameters t
+    isempty(species) && (species = [funcsym(:S,t,i) for i = 1:numspecs])
+
+    rn = Vector{Reaction}(undef,numrxs)
+    sub_indices = argmin(incidencemat, dims=1)     # cartesian indices of substrate complexes
+    prod_indices = argmax(incidencemat, dims=1)    # cartesian indicies of products complexes
+
+    for i ∈ 1:numrxs
+        subStoichInd = getindex.(findall(!iszero, compstoichmat[:,sub_indices[i][1]]))
+        prodStoichInd = getindex.(findall(!iszero, compstoichmat[:,prod_indices[i][1]]))
+
+        if subStoichInd == Int64[] &&  prodStoichInd != Int64[]
```

Comment:
Why not just check `if isempty(subStoichInd) && !isempty(prodStoichInd)`?

Also, variables should be in all lower case in Julia, so you should use `substoichind` or `sub_stoich_ind`.

Reply:
done!